File: React-Native/PrimeiroProjeto/android/app/src/main/java/com/primeiroprojeto/MathModule.java
package com.primeiroprojeto;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

import javax.annotation.Nonnull;

public class MathModule extends ReactContextBaseJavaModule {

    public MathModule(@Nonnull ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @ReactMethod
    public void sum(int numA, int numB, Promise promise) {
        int result = numA + numB;
        promise.resolve(result);
    }

    @Nonnull
    @Override
    public String getName() {
        return "Math";
    }

}

File: React-Native/PrimeiroProjeto/ios/PrimeiroProjeto/MathModule.m
//
//  MathModule.m
//  PrimeiroProjeto
//
//  Created by Douglas Nassif Roma Junior on 06/09/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import "MathModule.h"

#import <React/RCTConvert.h>

@implementation MathModule

RCT_EXPORT_MODULE(Math);

RCT_EXPORT_METHOD(sum: (NSInteger) numA
                  with: (NSInteger) numB
                  withResolver: (RCTPromiseResolveBlock) resolve
                  andRejecter: (RCTPromiseRejectBlock) reject)
{
  dispatch_sync(nil, ^{
    long result = numA + numB;
    resolve([NSNumber numberWithLong:result]);
  });
}

@end
